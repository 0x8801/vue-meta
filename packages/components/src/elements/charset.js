import { createComponent, variantProps } from '../utils'

export const charset = {
  functional: true,
  props: {
    value: String
  },
  render(h, { data, props }) {
    data.attrs = {
      ...data.attrs,
      charset: props.value
    }

    return h('meta', data)
  }
}
