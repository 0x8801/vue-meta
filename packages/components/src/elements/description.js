import { createComponent, variantProps } from '../utils'

export function createDescriptionComponent({ namePrefix = '', nameAttribute = 'name' } = {}) {
  return createComponent({
    valueAttribute: 'content',
    attrs: {
      [nameAttribute]: `${namePrefix}${namePrefix ? ':' : ''}description`
    }
  })
}

export const htmlDescription = createDescriptionComponent()

export const ogDescription = createDescriptionComponent({ namePrefix: 'og' })

export const twitterDescription = createDescriptionComponent({ namePrefix: 'twitter' })

export const websiteDescription = createDescriptionComponent({ nameAttribute: 'itemprop' })

export const variants = {
  html: htmlDescription,
  og: ogDescription,
  twitter: twitterDescription,
  website: websiteDescription
}

export const description = {
  ...createComponent(),
  render(h, { data, props }) {
    return Object.keys(variantProps)
      .filter(variant => props[variant] && !!variants[variant])
      .map(variant => h(variants[variant], {
        ...data,
        props: { ...props }
      }))
  }
}
