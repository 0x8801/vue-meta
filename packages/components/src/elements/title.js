import { createComponent, variantProps } from '../utils'

function createTitleComponent({ tag = 'meta', namePrefix = '', nameAttribute = 'name' } = {}) {
  const isHtmlTitle = tag === 'title'

  return createComponent({
    tag,
    valueAttribute: isHtmlTitle ? undefined : 'content',
    attrs: isHtmlTitle ? undefined : {
      [nameAttribute]: `${namePrefix}${namePrefix ? ':' : ''}title`
    }
  })
}

export const htmlTitle = createTitleComponent({ tag: 'title' })

export const ogTitle = createTitleComponent({ namePrefix: 'og' })

export const twitterTitle = createTitleComponent({ namePrefix: 'twitter' })

export const websiteTitle = createTitleComponent({ nameAttribute: 'itemprop' })

export const variants = {
  html: htmlTitle,
  og: ogTitle,
  twitter: twitterTitle,
  website: websiteTitle
}

export const title = {
  ...createComponent(),
  render(h, { data, props }) {
    return Object.keys(variantProps)
      .filter(variant => props[variant] && !!variants[variant])
      .map(variant => h(variants[variant], {
        ...data,
        props: { ...props }
      }))
  }
}
