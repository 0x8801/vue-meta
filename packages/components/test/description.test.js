import { renderToString } from '@vue/server-test-utils'
import * as descriptions from '../src/elements/description'

const renderOptions = {
  propsData: {
    value: 'test'
  }
}

test('description', async () => {
  const str = await renderToString(descriptions.htmlDescription, renderOptions)
  expect(str).toMatchSnapshot()
})

test('og-description', async () => {
  const str = await renderToString(descriptions.ogDescription, renderOptions)
  expect(str).toMatchSnapshot()
})

test('twitter-description', async () => {
  const str = await renderToString(descriptions.twitterDescription, renderOptions)
  expect(str).toMatchSnapshot()
})

test('website-description', async () => {
  const str = await renderToString(descriptions.websiteDescription, renderOptions)
  expect(str).toMatchSnapshot()
})

test('description:html', async () => {
  const str = await renderToString(descriptions.description, {
    ...renderOptions,
    propsData: {
      ...renderOptions.propsData,
      html: true
    }
  })
  expect(str).toMatchSnapshot()
})

test('description:og', async () => {
  const str = await renderToString(descriptions.description, {
    ...renderOptions,
    propsData: {
      ...renderOptions.propsData,
      html: false,
      og: true
    }
  })
  expect(str).toMatchSnapshot()
})

test('description:twitter', async () => {
  const str = await renderToString(descriptions.description, {
    ...renderOptions,
    propsData: {
      ...renderOptions.propsData,
      html: false,
      twitter: true
    }
  })
  expect(str).toMatchSnapshot()
})

test('description:website', async () => {
  const str = await renderToString(descriptions.description, {
    ...renderOptions,
    propsData: {
      ...renderOptions.propsData,
      html: false,
      website: true
    }
  })
  expect(str).toMatchSnapshot()
})
