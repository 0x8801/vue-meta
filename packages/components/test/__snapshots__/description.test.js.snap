// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description 1`] = `"<meta name=\\"description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;

exports[`description:html 1`] = `"<meta name=\\"description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;

exports[`description:og 1`] = `"<meta name=\\"og:description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;

exports[`description:twitter 1`] = `"<meta name=\\"twitter:description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;

exports[`description:website 1`] = `"<meta itemprop=\\"description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;

exports[`og-description 1`] = `"<meta name=\\"og:description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;

exports[`twitter-description 1`] = `"<meta name=\\"twitter:description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;

exports[`website-description 1`] = `"<meta itemprop=\\"description\\" content=\\"test\\" data-server-rendered=\\"true\\">"`;
